python/pyiceberg/catalog/glue.py

```diff
@@ -0,0 +1,453 @@
+            )
+        except self.glue.exceptions.AlreadyExistsException as e:
+            raise TableAlreadyExistsError(f"Table {database_name}.{table_name} already exists") from e
+        except self.glue.exceptions.EntityNotFoundException as e:
+            raise NoSuchNamespaceError(f"Database {database_name} not found") from e
+
+        try:
+            loaded_table = self.load_table(identifier=(database_name, table_name))
+        except self.glue.exceptions.EntityNotFoundException as e:
+            raise NoSuchTableError(f"Table {database_name}.{table_name} fail to be created") from e
+
+        return loaded_table
+
+    def load_table(self, identifier: Union[str, Identifier]) -> Table:
+        """Loads the table's metadata and returns the table instance.
+        You can also use this method to check for table existence using 'try catalog.table() except TableNotFoundError'
+        Note: This method doesn't scan data stored in the table.
+        Args:
+            identifier: Table identifier.
+        Returns:
+            Table: the table instance with its metadata
+        Raises:
+            NoSuchTableError: If a table with the name does not exist, or the identifier is invalid
+        """
+        database_name, table_name = self.identifier_to_tuple(identifier)
+        try:
+            load_table_response = self.glue.get_table(DatabaseName=database_name, Name=table_name)
+        except self.glue.exceptions.EntityNotFoundException as e:
+            raise NoSuchTableError(f"Table does not exists: {database_name}.{table_name}") from e
+
+        return self._convert_glue_to_iceberg(load_table_response[PROP_GLUE_TABLE])
+
+    def drop_table(self, identifier: Union[str, Identifier]) -> None:
+        """Drop a table.
+
+        Args:
+            identifier: Table identifier.
+
+        Raises:
+            NoSuchTableError: If a table with the name does not exist, or the identifier is invalid
+        """
+        database_name, table_name = self.identifier_to_tuple(identifier)
+        try:
+            self.glue.delete_table(DatabaseName=database_name, Name=table_name)
+        except self.glue.exceptions.EntityNotFoundException as e:
+            raise NoSuchTableError(f"Table does not exists: {database_name}.{table_name}") from e
+
+    def purge_table(self, identifier: Union[str, Identifier]) -> None:
+        """Drop a table and purge all data and metadata files. AWS glue will asynchronously delete resource
+        belongs to the table once the table is deleted
+
+        Args:
+            identifier (str | Identifier): Table identifier.
+
+        Raises:
+            NoSuchTableError: If a table with the name does not exist
+        """
+        self.drop_table(identifier)
+
+    def rename_table(self, from_identifier: Union[str, Identifier], to_identifier: Union[str, Identifier]) -> Table:
+        """Rename a fully classified table name
```

Comment:
Same doubt here. This sentence is copied from the interface in `pyiceberg.catalog.__init__.py` to ensure consistency.

My guess is that "fully classified" means that in cases of hierarchical namespaces, all the parent namespaces of the table are given in the identifier. For example, if the structure is `db1/db2/table`, the identifier should be `(db1, db2, table)`. Since AWS glue does not support hierarchical namespaces, we shall worry about nothing here. 

By the way, this reminds me to add an additional check when parsing identifiers to invalidate hierarchical namespaces structure. I will add this check soon